modules/packages/ZMQ.chpl

```diff
@@ -0,0 +1,357 @@
+// ZMQ.version
+// libzmq version 4.0.*
+
+/*
+  Module ZMQ: Chapel bindings for the ZeroMQ Messaging Library
+*/
+module ZMQ {
+
+  require "zmq.h", "-lzmq";
+
+  /*
+    External ZMQ interface
+     - Currently, the special nature of Chapel strings means that we have to
+       choose between using pointer types (which can clear up C compilation
+       warnings) and strings.  This should go away with the new strings.
+     - Ideally, these would be "private" extern procs, which could only be
+       directly accessible inside module ZMQ.
+   */
+  private extern var errno: c_int;
+  // -- Types
+  extern type zmq_msg_t;
+  type zmq_msg_p = c_ptr(zmq_msg_t);
```

Comment:
Is it more consistent with other Chapel code to specify pointer arguments to C functions as `foo(arg: c_ptr(c_type))` or `foo(ref arg: c_type)`?